src/Product/WCProductAdapter.php

```diff
@@ -176,21 +177,85 @@ protected function map_wc_general_attributes() {
 			$this->setItemGroupId( $this->parent_wc_product->get_id() );
 		}
 
+		// set product type using merchants defined product categories
+		$base_product_id      = $this->is_variation() ? $this->parent_wc_product->get_id() : $this->wc_product->get_id();
```

Comment:
The code here can be moved to a new method (e.g. `map_product_categories` or `map_product_types`) to reduce the clutter here.